cybersyn/scripts/commands.lua

```diff
@@ -1,47 +1,106 @@
+			local comb_2 = nil
+			local depot  = nil
+			local refuel = nil
+
+			for _,c in pairs(s.find_entities_filtered {name="cybersyn-combinator", position=ts.position, radius=3}) do
+				local op = c.get_control_behavior()
+				op = op and op.parameters.operation
+
+				if op == MODE_PRIMARY_IO or op == MODE_PRIMARY_IO_ACTIVE or op == MODE_PRIMARY_IO_FAILED_REQUEST then
+					if not comb_1 then comb_1 = c else report(ts, {"cybersyn-problems.double-station"}) end
+				elseif op == MODE_SECONDARY_IO then
+					if not comb_2 then comb_2 = c else report(ts, {"cybersyn-problems.double-station-control"}) end
+				elseif op == MODE_DEPOT then
+					if not depot  then depot  = c else report(ts, {"cybersyn-problems.double-depot"}) end
+				elseif op == MODE_REFUELER then
+					if not refuel then refuel = c else report(ts, {"cybersyn-problems.double-refueler"}) end
+				end
+			end
+
+			if comb_1 and depot  then report(ts, {"cybersyn-problems.station-and-depot"}) end
+			if comb_1 and refuel then report(ts, {"cybersyn-problems.station-and-refueler"}) end
+			if depot  and refuel then report(ts, {"cybersyn-problems.depot-and-refueler"}) end
+
+			if comb_1 then -- station mode takes precedence
+				station_types [ts.unit_number] = MODE_PRIMARY_IO
+				station_names [ts.backer_name] = true
+			elseif depot then
+				station_types [ts.unit_number] = MODE_DEPOT
+				depot_names   [ts.backer_name] = true
+			elseif refuel then
+				station_types [ts.unit_number] = MODE_REFUELER
+				refueler_names[ts.backer_name] = true
 			end
 		end
 	end
-	return cybersyn_names
+
+	return station_types, station_names, depot_names, refueler_names
 end
 
---- Run a function on each stop with non-default priority
----@param callback fun(train_stop: LuaEntity) 
-local function for_each_stop_with_invalid_priority(callback)
-	-- Priority is only problematic when a station is named the same as a
-	-- Cybersyn station.
-	local cybersyn_names = find_cybersyn_station_names()
+--- @param report function(LuaEntity, LocalisedString)
+local function find_problems(report)
+	local types, stations, depots, refuelers = check_single_stations_and_collect_data(report)
+
+	-- global checks 
 	for _,s in pairs(game.surfaces) do
 		for _,ts in pairs(s.find_entities_filtered {name="train-stop"}) do
-			if ts.train_stop_priority ~= 50 and cybersyn_names[ts.backer_name] then
-				callback(ts)
+			-- priority is only problematic when a station is named the same as a Cybersyn requester/provider
+			local name = ts.backer_name
+			if ts.train_stop_priority ~= 50 and (stations[name] or depots[name] or refuelers[name]) then
+				report(ts, {"cybersyn-problems.non-default-priority"})
+			end
+
+			local type = types[ts.unit_number]
+			if type ~= MODE_DEPOT and depots[ts.backer_name] then
+				report(ts, {"cybersyn-problems.name-overlap-with-depot"})
+			end
+			if type ~= MODE_REFUELER and refuelers[ts.backer_name] then
+				report(ts, {"cybersyn-problems.name-overlap-with-refueler"})
 			end
 		end
 	end
```

Comment:
UX: Maybe print something if nothing is found so people don't think the command didn't work?

Reply:
done